app/controllers/aid_applications/applicants_controller.rb

```diff
@@ -13,7 +13,9 @@ def update
       @aid_application = current_aid_application
       @aid_application.assign_attributes(aid_application_params)
 
-      if params[:form_action] == 'submit'
+      app_is_duplicate = AidApplication.matching_submitted_apps(@aid_application).any?
```

Comment:
One thing to keep in mind here is if any modifications happen as part of the validation (e.g. auto_strip_attributes) they won't have been triggered at this point. 

I wonder if it would be better to only invoke this within the `if form_action == 'submit'` block so that it's only invoked on the specific submission step. I also wonder if it would be good to call `save` (without a context) on submit before doing the submit validation check/save context. 

...which then makes me think maybe this check should be inside of `save_and_submit` so that it's wrapped up in the model itself rather than being a side effect of the controller. and maybe even a validation of that step... but I bet you've considered all that. 

I think simplest thing would be to just strip attributes as part of `matching_submitted_apps`